File: merchantdemo/src/main/java/com/example/merchantdemo/service/UserService.java
package com.example.merchantdemo.service;

import com.example.merchantdemo.dto.ChangePasswordDto;
import com.example.merchantdemo.dto.ResetPasswordDto;
import com.example.merchantdemo.dto.LoginUserDto;
import com.example.merchantdemo.entity.User;
import org.springframework.http.ResponseEntity;

import java.util.List;

public interface UserService {
    ResponseEntity<User> findByName(String theName);

    ResponseEntity<String> addUser(User theUser);

    ResponseEntity<List<User>> findAll();

    ResponseEntity<String> loginUser(LoginUserDto loginUser);


   // String resetPassword(Question question, String userName);



    ResponseEntity<Integer> findUserIdByUserName(String theName);

    ResponseEntity<User> findUserByUserId(int theId);

    ResponseEntity<String> changePassword(ChangePasswordDto theChangePassword);

    ResponseEntity<String> resetPassword(ResetPasswordDto theResetPassword);
}

File: merchantFrontEnd/src/Service/UserService.js
import axios from "axios";

const USER_API_BASE_URL = "http://localhost:8080/user";
class UserService {
  loginUser(loginUser) {
    return axios.post(USER_API_BASE_URL + "/login", loginUser);
  }

  findUserIdByUserName(theName) {
    return axios.get(USER_API_BASE_URL + "/userName/" + theName);
  }

  findUserByUserId(theId) {
    return axios.get(USER_API_BASE_URL + "/userId/" + theId);
  }
  addUser(user) {
    return axios.post(USER_API_BASE_URL + "/signup", user);
  }

  findByName(theName) {
    return axios.get(USER_API_BASE_URL + "/" + theName);
  }

  changePassword(theChangePassword) {
    return axios.put(USER_API_BASE_URL + "/changePassword", theChangePassword);
  }

  resetPassword(theResetPassword) {
    return axios.put(USER_API_BASE_URL + "/resetPassword", theResetPassword);
  }
}

export default new UserService();
